### PR TITLE
ClauseFeatureExtractor now works with full parses

### PR DESCRIPTION
**Pull request by christos-c**
Fixes an issue where the user would not create a PSEUDO_PARSE `View` before extracting features (while the full parse existed).

### CODE REVIEW

File: edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java [hunk 1]
```diff
@@ -45,7 +45,11 @@ public ClauseFeatureExtractor(String pseudoParseViewName, String clauseViewName)
     @Override
     public Set<Feature> getFeatures(Constituent c) throws EdisonException {
         TextAnnotation ta = c.getTextAnnotation();
-        TreeView pseudoParseView = (TreeView) ta.getView(parseViewName);
+        TreeView pseudoParseView;
+        // If the real parse view exists, use that
```
Comment (danyaljj) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: could you add a little clarification here why the pseudo-parse might not exist, while the real parse exists?

Comment (christos-c) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: done

File: edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java [hunk 2]
```diff
@@ -45,7 +45,14 @@ public ClauseFeatureExtractor(String pseudoParseViewName, String clauseViewName)
     @Override
     public Set<Feature> getFeatures(Constituent c) throws EdisonException {
         TextAnnotation ta = c.getTextAnnotation();
-        TreeView pseudoParseView = (TreeView) ta.getView(parseViewName);
+        TreeView pseudoParseView;
+        // If the real parse view exists, use that.
+        // Normally, this feature requires the PSEUDO_PARSE to be manually created by the user
+        // as it is cheaper to make than a full parse. However, there are cases
+        // where the user has already created a full parse view (for other features)
```
Comment (danyaljj) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: I'm confused now; I thought we build pseudo-parse automatically based on other annotations (e.g. real parse): https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/blob/0965deae01ff6bcd176f96b9fd050369f0ae3e1e/edison/src/main/java/edu/illinois/cs/cogcomp/edison/annotators/PseudoParse.java

Comment (christos-c) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: No, it's based on chunks and POS tags.

Comment (danyaljj) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: ok POS and/or chunks might not be available, in which case we would use the real parse ...

Comment (christos-c) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: Yes, but in general it should be up to the user to decide whether they will add the PSEUDO_PARSE or the real parse view (and their required views) and this feature should be able to use either.

Comment (danyaljj) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: > but in general it should be up to the user to decide whether they will add the PSEUDO_PARSE or the real parse  

Agreed; maybe add a TODO to make it parameterized? (i.e. the target view as parameter)

Comment (christos-c) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: The target view *is* parameterised! It's just that the default constructor use the PSEUDO_PARSE views.

Comment (danyaljj) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: got it; makes sense ...

**Comment by danyaljj** [comment]
Merging on green.

**Pull request merged**